enterprise/backend/src/metabase_enterprise/transforms/query_impl.clj

```diff
@@ -13,5 +21,71 @@
   [transform]
   (-> transform :source :query :database))
 
+(mr/def ::transform-details
+  [:map
+   [:transform-type [:enum {:decode/normalize schema.common/normalize-keyword} :table]]
+   [:conn-spec :any]
+   [:query :string]
+   [:output-table [:keyword {:decode/normalize schema.common/normalize-keyword}]]])
+
+(mr/def ::transform-opts
+  [:map
+   [:overwrite? :boolean]])
+
+(defn- transform-opts [{:keys [transform-type]}]
+  (case transform-type
+    :table {:overwrite? true}
+
+    ;; once we have more than just append, dispatch on :target-incremental-strategy
+    :table-incremental {}))
+
+(defn run-mbql-transform!
+  "Run `transform` and sync its target table.
+
+  This is executing synchronously, but supports being kicked off in the background
+  by delivering the `start-promise` just before the start when the beginning of the execution has been booked
+  in the database."
+  ([transform] (run-mbql-transform! transform nil))
+  ([{:keys [id source target] :as transform} {:keys [run-method start-promise]}]
+   (try
+     (let [db (get-in source [:query :database])
+           {driver :engine :as database} (t2/select-one :model/Database db)
+           transform-details {:db-id db
+                              :database database
+                              :transform-id   id
+                              :transform-type (keyword (:type target))
+                              :conn-spec (driver/connection-spec driver database)
+                              :query (transforms.util/compile-source transform)
+                              :output-schema (:schema target)
+                              :output-table (transforms.util/qualified-table-name driver target)}
```

Comment:
Since we're moving this code anyway, seems like a good time to fix alignment etc 😁